java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java

```diff
@@ -58,19 +58,20 @@ public SurefireJavaParser(ResourcePerspectives perspectives, JavaResourceLocator
     this.javaResourceLocator = javaResourceLocator;
   }
 
-
-  public void collect(SensorContext context, File reportsDir) {
-    File[] xmlFiles = getReports(reportsDir);
+  public void collect(SensorContext context, File reportsDir, boolean reportDirSetByUser) {
```

Comment:
In order to avoid a method with a flag parameter, I would prefer to have 2 public methods such has "collectFromUserReportDir" and "collectFromDefault" for instance, and make this method private.